File: Lab3/app/src/main/java/com/example/pamo/lab3/fragment/QuizSummaryFragment.java
package com.example.pamo.lab3.fragment;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import com.example.pamo.lab3.R;

public class QuizSummaryFragment extends Fragment {


    private static final String SCORE_PARAM = "score";
    private static final String QUESTIONS_COUNT_PARAM = "questionsCount";

    private int score;
    private int questionsCount;

    public QuizSummaryFragment() {
    }

    public static QuizSummaryFragment newInstance(int score, int questionsCount) {
        QuizSummaryFragment fragment = new QuizSummaryFragment();
        Bundle args = new Bundle();
        args.putInt(SCORE_PARAM, score);
        args.putInt(QUESTIONS_COUNT_PARAM, questionsCount);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            score = getArguments().getInt(SCORE_PARAM);
            questionsCount = getArguments().getInt(QUESTIONS_COUNT_PARAM);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_quiz_summary, container, false);

        TextView summaryView = view.findViewById(R.id.quiz_summary);
        summaryView.setText(getString(R.string.t_summary_score, score, questionsCount));

        Button resetButton = view.findViewById(R.id.btn_quiz_reset);
        resetButton.setOnClickListener(resetListener);
        return view;
    }

    private final OnClickListener resetListener = v -> {
        QuizFragment quizFragment = (QuizFragment) requireActivity()
                .getSupportFragmentManager()
                .findFragmentByTag("QUIZ_FRAGMENT");

        if (quizFragment != null) {
            requireActivity().getSupportFragmentManager()
                    .popBackStackImmediate();
            quizFragment.startQuiz();
        }
    };
}

File: Lab4/app/src/main/java/com/example/pamo/lab4/fragment/QuizSummaryFragment.kt
package com.example.pamo.lab4.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.example.pamo.lab4.R

class QuizSummaryFragment : Fragment() {

    private var score = 0
    private var questionsCount = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            score = requireArguments().getInt(SCORE_PARAM)
            questionsCount = requireArguments().getInt(QUESTIONS_COUNT_PARAM)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_quiz_summary, container, false)
        val summaryView = view.findViewById<TextView>(R.id.quiz_summary)
        summaryView.text = getString(R.string.t_summary_score, score, questionsCount)
        val resetButton = view.findViewById<Button>(R.id.btn_quiz_reset)
        resetButton.setOnClickListener(resetListener)
        return view
    }

    private val resetListener = View.OnClickListener { v: View? ->
        val quizFragment = requireActivity()
                .supportFragmentManager
                .findFragmentByTag("QUIZ_FRAGMENT") as QuizFragment?
        if (quizFragment != null) {
            requireActivity().supportFragmentManager
                    .popBackStackImmediate()
            quizFragment.startQuiz()
        }
    }

    companion object {
        private const val SCORE_PARAM = "score"
        private const val QUESTIONS_COUNT_PARAM = "questionsCount"
    }
}